account_invoice_date_due/models/account_move.py

```diff
@@ -17,6 +17,22 @@ def _onchange_invoice_date_due_payment_term(self):
         if self.invoice_date_due_payment_term:
             self.invoice_date_due = self.invoice_date_due_payment_term
 
+    def _compute_amount(self):
+        # Avoid recomputation of amount fields when we are only changing the
+        # invoice's due date
+        if self.env.context.get("bypass_compute_amount"):
+            return
+        return super()._compute_amount()
+
+    def onchange(self, values, field_name, field_onchange):
+        obj = self
+        if field_name == "invoice_date_due":
+            # A change in the invoice date due would trigger a recompute of the
```

Comment:
Should we constraint this to the `posted` state?

Reply:
I was just writing a question about this :smile: 

I had some issues due to the fact that the change of the date due after the invoice had been posted triggered a recompute of the amounts in the lines and was marking the invoice as paid. I don't know if this is the correct fix though, as what I observed was that the residual amount was being computed as 0.0 after the change of the due date (See https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L1316)

The constraint you mention makes sense, though, if we follow this approach.

Reply:
I'd say so. Otherwise you're blocking the expected standard behavior. After that, as in core it's not possible to change the due date from the invoice, we must prevent to trigger that recompute: https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L491

Reply:
Ok, I made the suggested change of limiting to the `posted` state.

I also pushed another commit to fix the tests of `account_billing`, that were failing due to the change in Odoo from a couple of weeks ago of limiting the journal types for the specific invoice types.